pkg/rules/engine/engine.go

```diff
@@ -67,6 +72,25 @@ func NewEngine(sigs []detect.Signature, sources EventSources, output chan detect
 	return &engine, nil
 }
 
+// StartPipeline receives an input chanel, and returns an output channel
+// allowing the rule engine to be used in the events pipeline
+func StartPipeline(cfg Config, input chan protocol.Event, done chan bool) <-chan detect.Finding {
+	output := make(chan detect.Finding)
+
+	source := EventSources{Tracee: input}
+	engine, err := NewEngine(cfg.Signatures, source, output, cfg)
```

Comment:
Yup, it is definitely weird, and not what we want going forward, but for this PR it was chosen because I didn't want to change even more things on this PR, and we will want to support the old experience for now, this will change in the upcoming weeks, allowing us to consolidate in one solution.

Reply:
Not saying we shouldn't support the old implementation, just saying we should refactor all the argument to be passed in one struct. The `tracee-rules` binary shouldn't be broken for this, just some code needs to move around. I think this is the kind of moving around which fits to the scope of this PR.

Reply:
Agreed. Instead of cfg.Signatures and cfg, right ? Its an easy change.